### PR TITLE
Create working multistage Dockerfile

### PR DESCRIPTION
**Pull request by jbowen93**
## Description

Create a new Dockerfile that meets the expectations of both the docker-compose and k8s cluster setups. 

This means that the entrypoint to the container is the binary itself (`celestia-appd` in this case) . 

It is expected that the appropriate arguments are passed to the container as if one was running the binary locally. Configuration isn't baked into the image and is expected to be passed as flags or using volume mounts. 

For examples see [docker-compose](https://github.com/celestiaorg/devops-test/blob/main/devnet/docker/docker-compose.yml#L7) and [k8s](https://github.com/celestiaorg/devops-test/blob/main/devnet/k8s/core-nodes/0/core0-deployment.yaml#L37).

This has been manually tested using the following instructions:
```
cd celestia-app && docker build -f docker/Dockerfile -t jbowen93/celestia-app:multi .
cd devops-test/devnet && sed -i 's/jbowen93\/celestia-app:devnet/jbowen93\/celestia-app:multi/g' docker/docker-compose.yml
docker-compose -f docker/docker-compose.yml up
```
as well as with the `devops-test/devnet/k8s` cluster

closes: #153 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by Bidon15 — APPROVED** [review]
🚢 
Awesome job! The devops-test compose file looks slick and sick!

**Review by evan-forbes — APPROVED** [review]
awesome utACK

File: docker/priv_validator_state.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "height": "0",
+    "round": 0,
+    "step": 0
+  }
```
Comment (adlerjohn) [inline docker/priv_validator_state.json:5]: nit
```suggestion
}

```

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,41 +1,36 @@
...
-USER cappuser
-ENTRYPOINT [ "/capp/entry.sh" ]
+
+# This allows us to always set the --home directory using an env 
+# var while still capturing all argumenmts passed at runtime
+ENTRYPOINT [ "/bin/bash", "-c", "exec ./celestia-appd \
+            --home ${HOME} \
+            \"${@}\"", "--" ]
+# Default command to run if no arguments are passed
+CMD ["--help"]
```
Comment (adlerjohn) [inline docker/Dockerfile:36]: Nit

```suggestion
CMD ["--help"]

```

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,41 +1,36 @@
...
+# p2p, rpc and prometheus port
 EXPOSE 26656 26657 1317 9090
-ADD /docker/entry.sh /capp/entry.sh
-USER root
-RUN chmod +x /capp/entry.sh
-USER cappuser
-ENTRYPOINT [ "/capp/entry.sh" ]
+
+# This allows us to always set the --home directory using an env 
+# var while still capturing all argumenmts passed at runtime
```
Comment (renaynay) [inline docker/Dockerfile:31]:
```suggestion
# var while still capturing all arguments passed at runtime
```

**Review by renaynay — COMMENTED** [review]
Since this Dockerfile is multi-stage and meets the requirements of k8s/compose setup, can we name it something ____.Dockerfile to indicate that this is its main use?

**Comment by jbowen93** [comment]
> Since this Dockerfile is multi-stage and meets the requirements of k8s/compose setup, can we name it something ____.Dockerfile to indicate that this is its main use?

My intention was that this would be the canonical and ideally only Dockerfile needed for this repository. As such I believe we should use the "top level" Dockerfile naming.

**Review by adlerjohn — APPROVED** [review]
utACK

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**